### PR TITLE
Add basic functionality of Remind command for v1.2

### LINKED ISSUE #61: RemindCommand - New Command
Add a new feature to remind the user of orders that are reaching near the deadline.
Default reminder days are 3 days from the current date.

### LINKED ISSUE #63: RemindCommand - Update UG and include Unit Testing
Include Remind Command in user guide
Add unit testing for class RemindCommand.

### PR DESCRIPTION
**Pull request by RuiXiong2211**
Basic functionality of Remind command.
User guide updated.
Unit tests added.

Ready to merge.

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -9,5 +9,7 @@
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    public static final String MESSAGE_PERSONS_REMINDER_OVERVIEW = "REMINDER! \n"
+            + "%1$d persons with an upcoming delivery date!";
```
Comment (xiinweii98) [inline src/main/java/seedu/address/commons/core/Messages.java]: "orders" ? instead of "persons"

File: src/main/java/seedu/address/logic/commands/RemindCommand.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.ReminderDatePredicate;
+
+/**
+ * Finds and lists all persons in CakeCollate database whose delivery date is within 3 days of the current date.
```
Comment (xiinweii98) [inline src/main/java/seedu/address/logic/commands/RemindCommand.java]: Same thing here, "orders"

**Review by xiinweii98 — COMMENTED** [review]
Nothing much to comment on the code as of now. Something to note though.. Add in a String MESSAGE_USAGE to your remind command so that your PR will be compatible with @pavz02 's HelpCommand. So depending on whose PR gets merged first, @pavz02 's HelpCommand#addAllCommands() needs a little editing.

**Comment by simran-bhadani3** [comment]
Apart from the replacement of person by order as already mentioned by @xiinweii98, LGTM as of now.

File: src/main/java/seedu/address/commons/core/Messages.java (hunk 1 above)
Comment (RuiXiong2211) [reply src/main/java/seedu/address/commons/core/Messages.java]: fixed

File: src/main/java/seedu/address/logic/commands/RemindCommand.java (hunk 2 above)
Comment (RuiXiong2211) [reply src/main/java/seedu/address/logic/commands/RemindCommand.java]: fixed

File: src/main/java/seedu/address/logic/commands/ClearCommand.java [hunk 3]
```diff
@@ -11,7 +11,8 @@
 public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
-    public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_SUCCESS = "CakeCollate has been cleared!";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Clears all entries from CakeCollate.";
```
Comment (xiinweii98) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java:15]: :o why is this here? Thought it is supposed to be changed by Pavitra's PR

File: src/main/java/seedu/address/logic/commands/RemindCommand.java [hunk 4]
```diff
@@ -7,14 +7,19 @@
 import seedu.address.model.person.ReminderDatePredicate;
 
 /**
- * Finds and lists all persons in CakeCollate database whose delivery date is within 3 days of the current date.
+ * Finds and lists all orders in CakeCollate database whose delivery date is within 3 days of the current date.
```
Comment (xiinweii98) [inline src/main/java/seedu/address/logic/commands/RemindCommand.java:10]: If you're changing such that user can choose the range of dates then you should remove "is within 3 days of the current date" right.

File: src/main/java/seedu/address/logic/parser/RemindCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+     * @throws ParseException if the user input does not conform the expected format
+     */
+
+    public RemindCommand parse(String args) throws ParseException {
+        String trimmedArg = args.trim();
+        if (trimmedArg.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemindCommand.MESSAGE_USAGE));
+        }
+        if (!StringUtil.isUnsignedInteger(trimmedArg)) {
```
Comment (xiinweii98) [inline src/main/java/seedu/address/logic/parser/RemindCommandParser.java:27]: You can maybe merge this two if you are not trying to differentiate the reason why there is a parse failure (since you are throwing the same exception). In a way the error message will be less meaningful but it is what the original coder(s) did too for AddCommandParser etc. We discussed this briefly during our last meeting when @pPris brought it up. I think for now we are KIV-ing and maybe changing it if we have time.

**Review by xiinweii98 — COMMENTED** [review]
Everything else looks good. Have fun adding tests lul.

File: src/main/java/seedu/address/model/person/ReminderDatePredicate.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Returns true if the order's delivery date is within X days of the current date.
+     */
+    public boolean isWithinXDays(Person person) {
+        LocalDate toTest = person.getDeliveryDate().getValue();
+        LocalDate dateToday = LocalDate.now();
+        LocalDate acceptableDate = dateToday.plusDays(days + 1);
+        return toTest.isEqual(dateToday) || toTest.isBefore(acceptableDate);
```
Comment (pavz02) [inline src/main/java/seedu/address/model/person/ReminderDatePredicate.java]: Isn't toTest.isBefore(acceptableDate) sufficient haha? Because dateToday is before the acceptable date? And Also because remind is for upcoming deadlines right, you could consider making sure all the dates are after today. In case a person/order with a past delivery date is still in the system.

File: src/main/java/seedu/address/logic/parser/RemindCommandParser.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.ReminderDatePredicate;
+
+/**
+ * Parses input argument and creates a new RemindCommand object
+ */
+public class RemindCommandParser implements Parser<RemindCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
```
Comment (pavz02) [inline src/main/java/seedu/address/logic/parser/RemindCommandParser.java]: What exactly does "in the context of the FindCommand" mean?

**Review by pavz02 — COMMENTED** [review]
Looks great!! 
Also, wanted to ask whether inputting an add/find command after a remind command changes the address book back to the original person list? Not sure if the addressbook already takes care of that somewhere else.

File: src/main/java/seedu/address/model/person/ReminderDatePredicate.java (hunk 6 above)
Comment (RuiXiong2211) [reply src/main/java/seedu/address/model/person/ReminderDatePredicate.java]: yep, changed the conditions to be more strict.

File: src/main/java/seedu/address/logic/parser/RemindCommandParser.java (hunk 7 above)
Comment (RuiXiong2211) [reply src/main/java/seedu/address/logic/parser/RemindCommandParser.java]: oops, meant to be RemindCommand

File: src/main/java/seedu/address/logic/commands/RemindCommand.java (hunk 4 above)
Comment (RuiXiong2211) [reply src/main/java/seedu/address/logic/commands/RemindCommand.java:10]: changed

File: src/main/java/seedu/address/model/person/ReminderDatePredicate.java (hunk 6 above)
Comment (RuiXiong2211) [reply src/main/java/seedu/address/model/person/ReminderDatePredicate.java]: > Looks great!!
> Also, wanted to ask whether inputting an add/find command after a remind command changes the address book back to the original person list? Not sure if the addressbook already takes care of that somewhere else.

@pavz02 , RemindCommand acts on a different list than add command, but on the same "filtered" order list as FindCommand, it will not affect the state of the original person list.

**Review by xiinweii98 — COMMENTED** [review]
LGTM 👍
